code/modules/mob/living/carbon/xenomorph/life.dm

```diff
@@ -44,10 +44,12 @@
 /mob/living/carbon/xenomorph/handle_fire()
 	. = ..()
 	if(.)
+		if(resting && fire_stacks > 0)
+			adjust_fire_stacks(-1)	//Passively lose firestacks when not on fire while resting and having firestacks built up.
```

Comment:
IMO this should be sped up a little. -1 is as fast as being on fire and you have to be resting, plus you can't resist to shake it off any faster because that signal is woven into actual ignition. Coded it to happen via Life() all the time for xenos when not on fire then decided to check this PR just in case.
```suggestion
			adjust_fire_stacks(-2)	//Passively lose firestacks when not on fire while resting and having firestacks built up.
```

Reply:
Having fire stacks while not on fire, with this PR at least, should be a very rare case that needs some special circumstances to occur. Currently, jelly prevents igniting but not gaining fire stacks (which e.g. walking on fire causes), which is how this is seen so commonly. With this PR, if you can't be ignited / are fully resistant (like jelly does), you shouldn't get fire stacks. This means that the only way for this to happen is to a) not be fully resistant to fire b) not be on fire and c) be affected by something that increases fire stacks without igniting you. Said combination of events (should) only happen with e.g. splashing ethanol or simillar on a xenos (which, who does that?) and is very very rare. This passively slowly decaying when resting is effectively just a catch so you don't have it for forever if something like that tags you and doesn't follow up with fire anytime soon, if you want to get rid of it quicker you can always set yourself on fire in a controlled way which immediately allows you to burn away the stacks.

Reply:
> With this PR, if you can't be ignited / are fully resistant (like jelly does), you shouldn't get fire stacks.

Okay that was my main concern and in-game it appears to be working that way so this is fair enough then. Consider it resolved. Thanks for the detailed response.